webrtc/videoroom/lib/videoroom/pipeline.ex

```diff
@@ -74,69 +76,85 @@ defmodule VideoRoom.Pipeline do
 
   @impl true
   def handle_other({:new_peer, peer_pid, peer_type, display_name, ref}, ctx, state) do
-    send(peer_pid, {:new_peer, {:ok, state.max_display_num}, ref})
-
-    if Map.has_key?(ctx.children, {:endpoint, peer_pid}) do
-      Membrane.Logger.warn("Peer already connected, ignoring")
-      {:ok, state}
-    else
-      Membrane.Logger.info("New peer #{inspect(peer_pid)} of type #{inspect(peer_type)}")
-      Process.monitor(peer_pid)
-
-      tracks = new_tracks(peer_type)
+    participants_num =
+      state.endpoints
+      |> Map.values()
+      |> Enum.count(&(&1.type == :participant))
 
-      endpoint = Endpoint.new(peer_pid, peer_type, tracks, %{display_name: display_name})
+    cond do
```

Comment:
I would opt for dividing this function into several smaller chunks. It is too big for me already.